### PR TITLE
Feature/filter-slider

### PR DESCRIPTION
**Pull request by StephieTack**

### CODE REVIEW

File: components/EmotionList.js [hunk 1]
```diff
@@ -9,40 +9,107 @@ export default function EmotionList({
...
+      const canScrollRight =
+        tabsBox.scrollWidth > tabsBox.clientWidth + tabsBox.scrollLeft;
+      setShowLeftArrow(canScrollLeft);
+      setShowRightArrow(canScrollRight);
+    }
+  }
+
+  function handleScroll() {
+    updateArrowVisibility();
+  }
```
Comment (F-Kirchhoff) [inline components/EmotionList.js]: instead onf handleScroll you can pass updateArrowVisibility to the eventHandler prop.

File: components/EmotionList.js [hunk 2]
```diff
@@ -9,40 +9,107 @@ export default function EmotionList({
...
+              &#9654;
+            </StyledIconRight>
+          )}
+        </StyledWrapper>
+      </StyledDivWrapper>
+      <StyledAppliedInfo>
+        {selectedFilter ? (
+          <>
+            #{selectedFilter}
+            <StyledClearFilter onClick={() => setSelectedFilter("")}>
```
Comment (F-Kirchhoff) [inline components/EmotionList.js]: Please rename this that it includes "button" in the name.

File: components/EmotionList.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import styled from "styled-components";
```
Comment (F-Kirchhoff) [inline components/EmotionList.js]: You have now a lot of code in your EmotionList Component. Please split this up such that the filter display is its own component and is called by your page index.js next to the Emotionlist.

File: components/EmotionList.js [hunk 4]
```diff
@@ -52,30 +119,113 @@ export default function EmotionList({
...
+  padding: 8px 10px;
+  border: 1px solid #e0e1f0;
+  border-radius: 8px;
+  transition: background-color 0.3s ease;
+  opacity: 70%;
+  -webkit-tap-highlight-color: transparent;
+
+  ${({ $isSelected, $color }) =>
+    $isSelected &&
+    `
```
Comment (F-Kirchhoff) [inline components/EmotionList.js]: you can use the function `css` from styled-components here, that helps to write this nested css code.
```suggestion
    css`
```

**Review by F-Kirchhoff — CHANGES_REQUESTED** [review]
Hey,
this is great code and a rather tricky feature to implement. I only have one big concern with the PR: your component EmotionList displays way more than the name suggests. Please split your code into a Filter and List component and render both from the page component in index.js.

File: components/EmotionList.js (hunk 2 above)
Comment (StephieTack) [reply components/EmotionList.js]: I did. I was not sure because at least this are no buttons. It is existing in the li-tags.

**Review by ahohnsen — APPROVED** [review]
Very nice feature. ✌🏼

File: components/EmotionListAndFilter/useScreenSize.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect } from "react";
+
+export default function useScreenSize() {
+  const [screenSize, setScreenSize] = useState({
+    width: 0,
+    height: 0,
+  });
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
```
Comment (F-Kirchhoff) [inline components/EmotionListAndFilter/useScreenSize.js]: a useEffect is only run in the client, therefore you don't need to check if window does exist.

**Review by F-Kirchhoff — APPROVED** [review]
Fantastic, I think this is what a library would have done instead, but recreating it is great as well.

File: components/EmotionListAndFilter/useScreenSize.js (hunk 5 above)
Comment (F-Kirchhoff) [reply components/EmotionListAndFilter/useScreenSize.js]: usually, hooks are placed in a folder `lib/hooks/` not in the components folder.

File: components/EmotionListAndFilter/Filter.js [hunk 6]
```diff
@@ -0,0 +1,224 @@
+import styled, { css } from "styled-components";
```
Comment (F-Kirchhoff) [inline components/EmotionListAndFilter/Filter.js]: please move these files out into the components folder and remove this folder.

**Pull request merged**